src/snowflake/snowpark/table_function.py

```diff
@@ -321,9 +319,16 @@ def get_column_names_from_plan(plan: SnowflakePlan) -> List[str]:
 
 
 def _get_cols_after_explode_join(
-    func: _ExplodeFunctionCall, plan: SnowflakePlan
+    func: _ExplodeFunctionCall,
+    plan: SnowflakePlan,
+    select_statement: Optional[SelectStatement],
 ) -> Tuple[List, List]:
-    explode_col_type = plan.output_dict.get(func.col, [None])[0]
+    if select_statement:
+        plan = select_statement.select([func.col._named()]).snowflake_plan
+        explode_col_type = plan.output[0].datatype
+    else:
+        col_name = func.col._expression.name
+        explode_col_type = plan.output_dict.get(col_name, [None])[0]
```

Comment:
previously the key used to retrieve info is of `Column` type and now it's a string, are they equivalent?

I searched the code base and it looks like output_dict is just using strings as keys, does that mean the previous `explode_col_type` could be wrong?

Reply:
I see, thanks for the explanation!